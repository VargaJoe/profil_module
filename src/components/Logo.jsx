import React, { Component } 		from 'react';
import logo 						from '../images/logo.png';
import { Link } 					from 'react-router-dom'

class Logo extends Component {
	render () {
		return (
			<div className="sidebar__logo">
				<Link to="/">
					<img src={logo} alt="site logo" />
				</Link>
			</div>
		);
	}
};

export default Logo;
