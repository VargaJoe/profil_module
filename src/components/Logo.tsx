import React				 		from 'react';
import logo 						from '../images/logo.png';
import { Link } 					from 'react-router-dom';

export interface MenuTrigger { 
	openMenu: boolean;
}

class Logo extends React.Component<MenuTrigger, {}> {
	constructor(props: MenuTrigger) {
        super(props);
	}
	
	render () {
		return (
			<div className="sn_sidebar__logo">
				<Link to="/">
					<img src={logo} alt="site logo" />
				</Link>
				<div className="sn_menu" onClick={this.props.openMenu}>
					<span></span>
				</div>
			</div>
		);
	}
};

export default Logo;
