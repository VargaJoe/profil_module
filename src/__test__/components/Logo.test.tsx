import * as React 						from 'react';
import Logo 							from '../../components/Logo';
import { 
	configure, 
    shallow,
    mount } 							from 'enzyme';
import * as Adapter 					from 'enzyme-adapter-react-16';
import toJson 				      		from 'enzyme-to-json';

configure( {adapter: new Adapter()} );

describe('<Logo /> shallow rendering', () => {
    
    const logo = shallow(<Logo />); 
    
    // test Snapshot 
	it('Match to snapshot', () => {
		expect(toJson(logo)).toMatchSnapshot();
    });

    it('Contain one menu element ', () => {
		expect(logo.find('.sn_menu').length).toBe(1);
    });

    it('Contain one img element ', () => {
		expect(logo.find('img').length).toBe(1);
    });
    
});
