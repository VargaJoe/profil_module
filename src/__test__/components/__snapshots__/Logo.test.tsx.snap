// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> shallow rendering Match to snapshot 1`] = `
<div
  className="sn_sidebar__logo"
>
  <Link
    replace={false}
    to="/"
  >
    <img
      alt="site logo"
      src="logo.png"
    />
  </Link>
  <div
    className="sn_menu"
  >
    <span />
  </div>
</div>
`;
